app/models/ability.rb

```diff
@@ -23,77 +23,77 @@ def initialize(user)
-      team.students.include?(user)
+      team.students.include?(user) && signed_in_and_confirmed?(user)
     end
 
     cannot :create, Team do |team|
       on_team_for_season?(user, team.season) || !user.confirmed?
     end
 
     can :join, Team do |team|
-      team.helpdesk_team? and signed_in?(user) and user.confirmed? and not on_team?(user, team)
+      team.helpdesk_team? and signed_in_and_confirmed?(user) and user.confirmed? and not on_team?(user, team)
     end
 
     can :crud, Role do |role|
-      user.admin? || on_team?(user, role.team)
+      (user.admin? || on_team?(user, role.team)) && signed_in_and_confirmed?(user)
     end
 
     can :crud, Source do |repo|
-      user.admin? || on_team?(user, repo.team)
+      (user.admin? || on_team?(user, role.team)) && signed_in_and_confirmed?(user)
     end
 
     can :supervise, Team do |team|
-      user.roles.organizer.any? || team.supervisors.include?(user)
+      (user.roles.organizer.any? || team.supervisors.include?(user)) && signed_in_and_confirmed?(user)
     end
 
     can :crud, ConferencePreference do |preference|
-      user.admin? || (preference.team.students.include? user)
+      (user.admin? || (preference.team.students.include? user)) && signed_in_and_confirmed?(user)
     end
 
-    can :crud, Conference if user.admin? || user.current_student?
+    can :crud, Conference if (user.admin? || user.current_student?) && signed_in_and_confirmed?(user)
 
     # todo add mailing controller and view for users in their namespace, where applicable
     can :read, Mailing do |mailing|
       mailing.recipient? user
     end
 
-    can :crud, :comments if user.admin?
-    can :read, :users_info if user.admin? || user.supervisor?
+    can :crud, :comments if (user.admin?) && signed_in_and_confirmed?(user)
+    can :read, :users_info if (user.admin? || user.supervisor?) && signed_in_and_confirmed?(user)
 
     # projects
     can :crud, Project do |project|
-      user.admin? ||
-        (user.confirmed? && user == project.submitter)
+      (user.admin? ||
+        (user.confirmed? && user == project.submitter)) && signed_in_and_confirmed?(user)
     end
     can :use_as_template, Project do |project|
-      user == project.submitter && !project.season&.current?
+      (user == project.submitter && !project.season&.current?) && signed_in_and_confirmed?(user)
     end
 
     can :create, Project if user.confirmed?
     cannot :create, Project if !user.confirmed?
 
     # activities
     can :read, :feed_entry
-    can :read, :mailing if signed_in?(user)
+    can :read, :mailing if signed_in_and_confirmed?(user)
 
     # applications
-    can :create, :application_draft if user.student? && user.application_drafts.in_current_season.none?
+    can :create, :application_draft if user.student? && user.application_drafts.in_current_season.none? && signed_in_and_confirmed?(user)
   end
 
-  def signed_in?(user)
-    user.persisted?
+  def signed_in_and_confirmed?(user)
```

Comment:
Now with reading all the changes, I think my previous suggestion for renaming this method was actually kinda misleading - sorry 🙈 

For the context of the abilities files - "signed in" as in whether or not there's a session is actually not relevant at all. If there's no user signed in, the call to the respective ability would not have a user object...

So I'd say it's fine to just call
```ruby
user.confirmed?
```
In the respective places (same as `user.admin?`)